addons/mail/static/src/messaging/entity/thread_viewer/thread_viewer.js

```diff
@@ -196,14 +230,22 @@ function ThreadViewerFactory({ Entity }) {
             });
 
             if (thread && this.thread !== thread) {
+                this._stopLoading();
                 this.link({ thread });
                 if (!this.threadCache.isLoaded && !this.threadCache.isLoading) {
                     this.threadCache.loadMessages();
                 }
             }
             if (this.threadCache !== prevThreadCache) {
+                this._stopLoading();
                 this._writeComponentHint('change-of-thread-cache');
             }
+
+            if (this.thread && this.threadCache.isLoading) {
+                if (!this._loaderTimeout) {
```

Comment:
Maybe move this condition inside `_prepareLoading`.

Reply:
A big wise master ( @alexkuhn -san) told me one day that he doesn't like function with a body starting with an `if` so I avoided it...

Reply:
Let's say what he has to say about it then. The reasoning is that we moved all the technical details about this timeout into those methods, it feels strange to not go all the way.

Reply:
Yep, this makes sense to me too

Reply:
The final word is ? :D

Reply:
Prevent leak of timeout:

1. either by changing the conditional expression (e.g. `!this.isShowingLoading && !this.isPreparingLoading`);
2. or move it inside `_prepareLoading` but then rename this function so that it makes sense to do nothing if a loading is already shown or in preparation.

I prefer 1st solution because `_prepareLoading()` is already smart enough from POV of users of this function, so it becomes though to find a clean naming for this function if we move the `if` statement in.